src/Request.ts

```diff
@@ -0,0 +1,350 @@
+import * as arrayUtil from 'dojo-core/array';
+import * as lang from 'dojo-core/lang';
+import Promise from 'dojo-core/Promise';
+import request, { Response, ResponsePromise } from 'dojo-core/request';
+import Filter from './Filter';
+import * as dstore from './interfaces';
+import QueryResults from './QueryResults';
+import Store, { NewableStoreModel } from './Store';
+
+const push = [].push;
+
+export interface RequestStoreArgs {
```

Comment:
This should presumably extend `StoreArgs` from `Store`, though it may ultimately be of little consequence depending on what we do about models.